tests/security/redprobe_secrets_gate.sh

```
@@ -0,0 +1,251 @@
+  esac
+done
+
+# Pinned by digest, matching security.yml. A movable tag would let the probe
+# and the gate drift apart silently.
+GITLEAKS_IMAGE='zricethezav/gitleaks:v8.30.1@sha256:c00b6bd0aeb3071cbcb79009cb16a60dd9e0a7c60e2be9ab65d25e6bc8abbb7f'
+
+command -v docker >/dev/null || { echo "docker is required" >&2; exit 2; }
+git rev-parse --verify "$REF" >/dev/null 2>&1 || { echo "no such ref: $REF" >&2; exit 2; }
+
+WORK="$(mktemp -d)"
+cleanup() { [ "$KEEP" -eq 1 ] && echo "kept: $WORK" || rm -rf "$WORK"; }
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)